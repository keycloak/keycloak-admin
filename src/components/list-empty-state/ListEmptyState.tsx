import React, { MouseEventHandler } from "react";
import {
  EmptyState,
  EmptyStateIcon,
  EmptyStateBody,
  Title,
  Button,
} from "@patternfly/react-core";
import { PlusCircleIcon } from "@patternfly/react-icons";

export type ListEmptyStateProps = {
  message: string;
  instructions: string;
  primaryActionText: string;
  onPrimaryAction: MouseEventHandler<HTMLButtonElement>;
};

export const ListEmptyState = ({
  message,
  instructions,
  primaryActionText,
  onPrimaryAction,
}: ListEmptyStateProps) => {
  return (
    <>
      <EmptyState variant="large">
        <EmptyStateIcon icon={PlusCircleIcon} />
        <Title headingLevel="h4" size="lg">
          {message}
        </Title>
        <EmptyStateBody>{instructions}</EmptyStateBody>
        <Button variant="primary" onClick={onPrimaryAction}>
          {primaryActionText}
        </Button>
      </EmptyState>
    </>
  );
};
