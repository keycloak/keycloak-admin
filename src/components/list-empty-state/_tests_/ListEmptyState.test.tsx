import React from "react";
import { render } from "@testing-library/react";
import { ListEmptyState } from "../ListEmptyState";

describe("<ListEmptyState />", () => {
  it("render", () => {
    const comp = render(
      <ListEmptyState
        message="No things"
        instructions="You haven't created any things for this list."
        primaryActionText="Add a thing"
        onPrimaryAction={() => {}}
      />
    );
    expect(comp.asFragment()).toMatchSnapshot();
  });
});
