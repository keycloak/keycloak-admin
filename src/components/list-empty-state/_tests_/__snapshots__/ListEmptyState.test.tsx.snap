// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListEmptyState /> render 1`] = `
<DocumentFragment>
  <div
    class="pf-c-empty-state pf-m-lg"
    data-testid="empty-state"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <svg
        aria-hidden="true"
        class="pf-c-empty-state__icon"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
        />
      </svg>
      <h4
        class="pf-c-title pf-m-lg"
      >
        No things
      </h4>
      <div
        class="pf-c-empty-state__body"
      >
        You haven't created any things for this list.
      </div>
      <button
        aria-disabled="false"
        class="pf-c-button pf-m-primary"
        data-ouia-component-id="OUIA-Generated-Button-primary-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        data-testid="no-things-empty-action"
        type="button"
      >
        Add it now!
      </button>
      <div
        class="pf-c-empty-state__secondary"
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-secondary"
          data-ouia-component-id="OUIA-Generated-Button-secondary-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          Add a thing
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
